test/react-web/client/graphql/queries.test.tsx

```diff
@@ -2176,6 +2176,38 @@ describe('queries', () => {
     expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
   });
 
+
+  it('will not try to refetch recycled `ObservableQuery`s when resetting the client store', () => {
+    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
+    const data = { allPeople: { people: [ { name: 'Luke Skywalker' } ] } };
+    const networkInterface = {
+      query: jest.fn(),
+    } as NetworkInterface;
+    const client = new ApolloClient({ networkInterface, addTypename: false });
+
+    @graphql(query)
+    class Container extends React.Component<any, any> {
+      render () {
+        return null;
+      }
+    }
+
+    const wrapper1 = renderer.create(
+      <ApolloProvider client={client}>
+        <Container/>
+      </ApolloProvider>
+    );
+
+    expect(Object.keys((client as any).queryManager.observableQueries)).toEqual(['1']);
```

Comment:
Could we check that this observable query still exists after we unmount as well?